src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1378,394 @@ public S(int arg)
+            var fieldKeywordSymbolInfo = speculativeModel.GetSymbolInfo(identifier);
+
+            Assert.Equal("System.Double C.<P>k__BackingField", comp.GetTypeByMetadataName("C").GetFieldsToEmit().Single().ToTestDisplayString());
+            Assert.Equal("System.Double C.<P>k__BackingField", fieldKeywordSymbolInfo.Symbol.ToTestDisplayString(includeNonNullable: true));
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Theory]
+        [InlineData("get => 0; get => field;", 19, false)]
+        [InlineData("get => field; get => 0;", 23, true)]
+        [InlineData("get { return 0; } get => field;", 27, false)]
+        [InlineData("get => field; get { return 0; }", 23, true)]
+        [InlineData("get => 0; get { return field; }", 19, false)]
+        [InlineData("get { return field; } get => 0;", 31, true)]
+        [InlineData("get { return 0; } get { return field; }", 27, false)]
+        [InlineData("get { return field; } get { return 0; }", 31, true)]
+        public void SpeculativeSemanticModel_FieldInDuplicateAccessorWhereFirstAccessorUsesField_BindOriginalFirst(string accessors, int diagnosticColumn, bool fieldAccessorFirst)
+        {
+            var comp = CreateCompilation($@"
+class C
+{{
+    public double P
+    {{
+        {accessors}
+    }}
+}}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            comp.VerifyDiagnostics(
+                // error CS1007: Property accessor already defined
+                Diagnostic(ErrorCode.ERR_DuplicateAccessor, "get").WithLocation(6, diagnosticColumn)
+                );
+
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken));
+
+            var model = comp.GetSemanticModel(tree);
+            var identifier = SyntaxFactory.ParseExpression("field");
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, (IdentifierNameSyntax)identifier, out var speculativeModel);
+
+            var fieldsToEmit = comp.GetTypeByMetadataName("C").GetFieldsToEmit();
+            if (fieldAccessorFirst)
+            {
+                Assert.Equal("System.Double C.<P>k__BackingField", fieldsToEmit.Single().ToTestDisplayString());
+            }
+            else
+            {
+                Assert.Empty(fieldsToEmit);
+            }
+
+            var fieldKeywordSymbolInfo = speculativeModel.GetSymbolInfo(identifier);
+            if (fieldAccessorFirst)
+            {
+                Assert.Equal("System.Double C.<P>k__BackingField", fieldKeywordSymbolInfo.Symbol.ToTestDisplayString(includeNonNullable: true));
```

Comment:
>Assert.Equal("System.Double C.<P>k__BackingField", fieldKeywordSymbolInfo.Symbol.ToTestDisplayString(includeNonNullable: true)); [](http://example.com/codeflow?start=16&length=128)

This is interesting. Somehow we are binding in context of a property accessor. I think we need to understand what exactly semantic model is doing.

Reply:
@AlekseyTs I'm not following here. Since the first accessor is using `field`, the backing field was marked as non-null calculated value. So we get a symbol back when speculating with `field`. Anything wrong with the current behavior?

Reply:
> Anything wrong with the current behavior?

Not so much wrong, but somewhat unexpected. We don't create symbols for duplicate accessors, but in order to get the field symbol we need to bind in context of an accessor. So, it feels like we are binding in context of an accessor that has different body. That is probably OK fall back strategy, but we need to confirm/understand exactly what is happening, whether that is an intentional fall back strategy or just an accident, etc. For example, does it mean that locals from the other accessor are in scope? I am not sure if we want this behavior.

Reply:
@AlekseyTs Thanks for clarifying. I'll investigate this tomorrow, it's a bit late here now :)

Reply:
Let me know if this behavior is okay so I can remove the PROTOTYPE comment.